planner/core/rule_column_pruning.go

```diff
@@ -63,32 +65,38 @@ func exprHasSetVarOrSleep(expr expression.Expression) bool {
 
 // PruneColumns implements LogicalPlan interface.
 // If any expression has SetVar function or Sleep function, we do not prune it.
-func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column) error {
+func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column, opt *logicalOptimizeOp) error {
 	child := p.children[0]
 	used := expression.GetUsedList(parentUsedCols, p.schema)
+	prunedColumns := make([]*expression.Column, 0)
```

Comment:
I guess it should be like this?

```suggestion
	prunedColumns := make([]*expression.Column, 0, 0)
```